### ISSUE TITLE
False positive for missing class docstring

### ISSUE THREAD

**Issue by pchanial** [PyCQA/pylint#3301]
### Steps to reproduce
```
class A:
    """The docstring for class A."""

class B(A):
    __doc__ = A.__doc__

```

### Current behavior
Class B triggers: C0115: Missing class docstring (missing-class-docstring)

### Expected behavior
No reporting

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
